### DISCUSSION TITLE
[svelte-leafletjs] - all callback seems invisible

### DISCUSSION THREAD

**Discussion by Ennoriel** [DefinitelyTyped/DefinitelyTyped#60263] [Issues with a @types package]
I don't know if it's an issue or me not knowing how to use svelte typescript declaration files.

I made a reproduction repository. Step for reproduction:
```bash
git clone git@github.com:Ennoriel/bug-types-svelte-leafletjs.git
npm ci
npm run check
```

result:
```bash
\bug-types-svelte-leafletjs\src\routes\index.svelte:8:12
Error: Type '{ getMap: () => Map; }' is not assignable to type 'IntrinsicAttributes & { options?: MapOptions | undefined; }'.
  Property 'getMap' does not exist on type 'IntrinsicAttributes & { options?: MapOptions | undefined; }'. (ts)               
                                                                                                                             
<LeafletMap bind:getMap />                                                                                     
```

It seems to me that:
```typescript
export class LeafletMap extends SvelteComponentTyped<
    {
        options?: L.MapOptions;
    },
    MapEvents,
    never
> {
    getMap(): L.Map;
}
```

should be:
```typescript
export class LeafletMap extends SvelteComponentTyped<
    {
        options?: L.MapOptions;
        getMap(): L.Map;
    },
    MapEvents,
    never
> { }
```

When svelte-kit is generating a package `.d.ts` file, it generates the callback in the props generics if it's a variable `let` and in the content of the interface if it's a const props this way:
```
{
    get callback(): void;
}
```
I don't know how the second one works though.

@k-yle I am tagging you as I think you might have a hint or are willing to know if it's a real issue.

[a comment by an automated account is not shown]

**Comment by k-yle** [comment]
Hi @Ennoriel, this is still the only svelte package on DefinitelyTyped, so it's quite possible that there are issues.

This is the pattern I'm used to:
```svelte
<script lang="ts">
  let map: LeafletMap;

  map.getMap();
</script>

<LeafletMap bind:this={map} />
```

The way you suggested works too, so I've created a PR to support both approaches. 

Please take a look at #60266, and if it works for you, please mark it as "Approved" in the GitHub UI so that it gets merged faster.

**Comment by Ennoriel** [comment]
@k-yle, Thanks for the initial PR and for the very quick fix PR!

I think that there is actually a prop missing: 

The component LeafletMap exposes an events prop which holds an event name list.

I think something like that should work:

```typescript
type KeyEvents = Array<keyof MapEvents>;

export class LeafletMap extends SvelteComponentTyped<
    {
        options?: L.MapOptions;
        events?: KeyEvents;
        getMap?(): L.Map;
    },
    MapEvents,
    never
    > {
    getMap(): L.Map;
}
```

I just saw that at the minunte. Sorry 😔

Also, side question here, I made a very quick types definition file for the svelte-table library locally. I'll ask the owner of the lib if he thinks to migrate the library to sveltkit/typescript and if not make a PR to DefinitelyTyped. If so, are you willing to help me reviewing the PR? I am not a Typescript expert.

**Reply by k-yle** [reply]
Sorry for the very delayed reply - You're right about `events`, I'll have time to fix this on the weekend. 

> If so, are you willing to help me reviewing the PR? I am not a Typescript expert.

Sure, feel free to @-mention me on any svelte PRs on DefinitelyTyped.
